sources/__tests__/catalog.test.ts

```diff
@@ -453,4 +453,60 @@ describe("yarn-plugin-catalogs", () => {
     const dependencies = extractDependencies(listOutput);
     expect(dependencies).includes("@use-funnel/core@npm:0.0.1");
   });
+
+  it("should not use default alias group if workspace is ignored", async () => {
+    workspace = await createTestWorkspace();
+
+    await workspace.writeYarnrc({
+      catalogs: {
+        options: {
+          default: ["root"],
+          ignoredWorkspaces: ["workspace-ignored"],
+        },
+        list: {
+          react: "npm:18.0.0",
+        },
+      },
+    });
+
+    await workspace.writeJson("package.json", {
+      name: "workspace-ignored",
+      version: "1.0.0",
+      private: true,
+      dependencies: {},
+    });
+
+    const { stderr } = await workspace.yarn.add("react");
+    const { stdout: listOutput } = await workspace.yarn.info();
+    const dependencies = extractDependencies(listOutput);
+
+    expect(stderr).toBe("");
+    expect(dependencies).not.includes("react@npm:18.0.0");
+  });
+
+  it("should fail validation if workspace is ignored, but using the catalog protocol", async () => {
+    workspace = await createTestWorkspace();
+
+    await workspace.writeYarnrc({
+      catalogs: {
+        options: {
+          ignoredWorkspaces: ["workspace-ignored"],
+        },
+        list: {
+          react: "npm:18.0.0",
+        },
+      },
+    });
+
+    await workspace.writeJson("package.json", {
+      name: "workspace-ignored",
+      version: "1.0.0",
+      private: true,
+      dependencies: {
+        react: "catalog:",
+      },
+    });
+
+    expect(workspace.yarn.install()).rejects.toThrow();
+  });
```

Comment:
Can we check that if it fails to do `yarn add react@catalog:stable` if the workspace is ignored?